### ISSUE TITLE
launch failed: cannot connect to the multipass socket

### ISSUE THREAD

**Issue by popey** [CanonicalLtd/multipass#655]
I just wanted to run snapcraft as a demo to someone and it failed with the error in the subject. I wonder if my change in network could be a factor. I put the laptop to sleep at home and woke it up in the London office (where I am today). I have a vague recollection that this happens when I sleep/wake and change network, but that may be a red herring.

```
snapcraft init
snapcraft
```

```
Launching a VM.
launch failed: cannot connect to the multipass socket                           
Please ensure multipassd is running and '/run/multipass_socket' is accessible
An error occurred when trying to launch the instance with 'multipass': returned exit code 2.
Ensure that 'multipass' is setup correctly and try again.

```

```
$ snap info multipass
name:      multipass
summary:   Ubuntu at your fingertips
publisher: Canonical✓
license:   unset
description: |
  Multipass gives you Ubuntu VMs in seconds. Just run `multipass launch` and it'll do all the setup
  for you. Check out `multipass find` for a list of available images. More details you can find
  under `multipass help`.
commands:
  - multipass
  - multipass.virsh
services:
  multipass.libvirt-bin: simple, enabled, inactive
  multipass.multipassd:  simple, enabled, inactive
snap-id:      mA11087v6dR3IEcQLgICQVjuvhUUBUKM
tracking:     beta
refresh-date: 6 days ago, at 08:07 GMT
channels:
  stable:    –                                       
  candidate: –                                       
  beta:      v0.5             2019-01-22 (587) 111MB classic
  edge:      v0.5-79-g75de7c4 2019-02-28 (629) 111MB classic
installed:   v0.5                        (587) 111MB classic

```

```
$ snap info snapcraft
snap inname:      snapcraft
summary:   easily create snaps
publisher: Canonical✓
contact:   snaps@canonical.com
license:   unset
description: |
  Package, distribute, and update any app for Linux and IoT.
  
  Snaps are containerised software packages that are simple to create and install. They auto-update
  and are safe to run. And because they bundle their dependencies, they work on all major Linux
  systems without modification.
commands:
  - snapcraft
snap-id:      vMTKRaLjnOJQetI78HjntT37VuoyssFE
tracking:     edge
refresh-date: 2 days ago, at 21:10 GMT
channels:
  stable:    3.2 2019-02-28 (2650) 57MB classic
  candidate: ↑                          
  beta:      ↑                          
  edge:      ↑                          
installed:   3.2            (2650) 57MB classic
```

```
$ snap get multipass
error: snap "multipass" has no configuration

```

```
$ journalctl -b0 -u snap.multipass.*
-- Logs begin at Mon 2018-10-01 20:07:09 BST, end at Fri 2019-03-01 11:13:40 GMT. --
Feb 28 15:22:52 KinkPad-K450 systemd[1]: Started Service for snap application multipass.libvirt-bin.
Feb 28 15:22:52 KinkPad-K450 systemd[1]: Started Service for snap application multipass.multipassd.
Feb 28 15:23:18 KinkPad-K450 multipassd[4898]: Unable to determine subnet for the mpqemubr0 subnet
Feb 28 15:23:18 KinkPad-K450 dnsmasq[8247]: started, version 2.75 cachesize 150
Feb 28 15:23:18 KinkPad-K450 dnsmasq[8247]: compile time options: IPv6 GNU-getopt DBus i18n IDN DHCP DHCPv6 no-Lua TFTP conntrack ipset
Feb 28 15:23:18 KinkPad-K450 dnsmasq-dhcp[8247]: DHCP, IP range 10.197.20.2 -- 10.197.20.254, lease time infinite
Feb 28 15:23:18 KinkPad-K450 dnsmasq-dhcp[8247]: DHCP, sockets bound exclusively to interface mpqemubr0
Feb 28 15:23:18 KinkPad-K450 dnsmasq[8247]: reading /etc/resolv.conf
Feb 28 15:23:18 KinkPad-K450 dnsmasq[8247]: using nameserver 127.0.0.53#53
Feb 28 15:23:18 KinkPad-K450 dnsmasq[8247]: read /etc/hosts - 7 addresses
Feb 28 15:23:18 KinkPad-K450 dnsmasq[8247]: cannot read /var/snap/multipass/common/data/multipassd/network/dnsmasq.hosts: No such file 
Feb 28 15:23:18 KinkPad-K450 multipassd[4898]: "No such interface 'org.freedesktop.DBus.Properties' on object at path /org/freedesktop/
Feb 28 15:23:18 KinkPad-K450 multipassd[4898]: Host cloud-images.ubuntu.com not found
Feb 28 15:23:19 KinkPad-K450 systemd[1]: snap.multipass.multipassd.service: Main process exited, code=exited, status=1/FAILURE
Feb 28 15:23:19 KinkPad-K450 systemd[1]: snap.multipass.multipassd.service: Failed with result 'exit-code'.
Feb 28 15:23:19 KinkPad-K450 systemd[1]: snap.multipass.multipassd.service: Service hold-off time over, scheduling restart.
Feb 28 15:23:19 KinkPad-K450 systemd[1]: snap.multipass.multipassd.service: Scheduled restart job, restart counter is at 1.
Feb 28 15:23:19 KinkPad-K450 systemd[1]: Stopped Service for snap application multipass.multipassd.
Feb 28 15:23:19 KinkPad-K450 systemd[1]: Started Service for snap application multipass.multipassd.
Feb 28 15:23:19 KinkPad-K450 multipassd[8277]: Unable to determine subnet for the mpqemubr0 subnet
Feb 28 15:23:19 KinkPad-K450 dnsmasq[8389]: started, version 2.75 cachesize 150
Feb 28 15:23:19 KinkPad-K450 dnsmasq[8389]: compile time options: IPv6 GNU-getopt DBus i18n IDN DHCP DHCPv6 no-Lua TFTP conntrack ipset
Feb 28 15:23:19 KinkPad-K450 dnsmasq-dhcp[8389]: DHCP, IP range 10.197.20.2 -- 10.197.20.254, lease time infinite
Feb 28 15:23:19 KinkPad-K450 dnsmasq-dhcp[8389]: DHCP, sockets bound exclusively to interface mpqemubr0
Feb 28 15:23:19 KinkPad-K450 dnsmasq[8389]: reading /etc/resolv.conf
Feb 28 15:23:19 KinkPad-K450 dnsmasq[8389]: using nameserver 127.0.0.53#53
Feb 28 15:23:19 KinkPad-K450 dnsmasq[8389]: read /etc/hosts - 7 addresses
Feb 28 15:23:19 KinkPad-K450 dnsmasq[8389]: cannot read /var/snap/multipass/common/data/multipassd/network/dnsmasq.hosts: No such file 
Feb 28 15:23:19 KinkPad-K450 multipassd[8277]: "No such interface 'org.freedesktop.DBus.Properties' on object at path /org/freedesktop/
Feb 28 15:23:19 KinkPad-K450 multipassd[8277]: Host cloud-images.ubuntu.com not found
Feb 28 15:23:19 KinkPad-K450 systemd[1]: snap.multipass.multipassd.service: Main process exited, code=exited, status=1/FAILURE
Feb 28 15:23:19 KinkPad-K450 systemd[1]: snap.multipass.multipassd.service: Failed with result 'exit-code'.
Feb 28 15:23:19 KinkPad-K450 systemd[1]: snap.multipass.multipassd.service: Service hold-off time over, scheduling restart.
Feb 28 15:23:19 KinkPad-K450 systemd[1]: snap.multipass.multipassd.service: Scheduled restart job, restart counter is at 2.
Feb 28 15:23:19 KinkPad-K450 systemd[1]: Stopped Service for snap application multipass.multipassd.
Feb 28 15:23:19 KinkPad-K450 systemd[1]: Started Service for snap application multipass.multipassd.
Feb 28 15:23:20 KinkPad-K450 multipassd[8438]: Unable to determine subnet for the mpqemubr0 subnet
Feb 28 15:23:20 KinkPad-K450 dnsmasq[8553]: started, version 2.75 cachesize 150
Feb 28 15:23:20 KinkPad-K450 dnsmasq[8553]: compile time options: IPv6 GNU-getopt DBus i18n IDN DHCP DHCPv6 no-Lua TFTP conntrack ipset
Feb 28 15:23:20 KinkPad-K450 dnsmasq-dhcp[8553]: DHCP, IP range 10.197.20.2 -- 10.197.20.254, lease time infinite
Feb 28 15:23:20 KinkPad-K450 dnsmasq-dhcp[8553]: DHCP, sockets bound exclusively to interface mpqemubr0
Feb 28 15:23:20 KinkPad-K450 dnsmasq[8553]: reading /etc/resolv.conf
Feb 28 15:23:20 KinkPad-K450 dnsmasq[8553]: using nameserver 127.0.0.53#53
Feb 28 15:23:20 KinkPad-K450 dnsmasq[8553]: read /etc/hosts - 7 addresses
Feb 28 15:23:20 KinkPad-K450 dnsmasq[8553]: cannot read /var/snap/multipass/common/data/multipassd/network/dnsmasq.hosts: No such file 
Feb 28 15:23:20 KinkPad-K450 multipassd[8438]: "No such interface 'org.freedesktop.DBus.Properties' on object at path /org/freedesktop/
Feb 28 15:23:20 KinkPad-K450 multipassd[8438]: Host cloud-images.ubuntu.com not found
Feb 28 15:23:20 KinkPad-K450 systemd[1]: snap.multipass.multipassd.service: Main process exited, code=exited, status=1/FAILURE
Feb 28 15:23:20 KinkPad-K450 systemd[1]: snap.multipass.multipassd.service: Failed with result 'exit-code'.
Feb 28 15:23:20 KinkPad-K450 systemd[1]: snap.multipass.multipassd.service: Service hold-off time over, scheduling restart.
Feb 28 15:23:20 KinkPad-K450 systemd[1]: snap.multipass.multipassd.service: Scheduled restart job, restart counter is at 3.
Feb 28 15:23:20 KinkPad-K450 systemd[1]: Stopped Service for snap application multipass.multipassd.
Feb 28 15:23:20 KinkPad-K450 systemd[1]: Started Service for snap application multipass.multipassd.
Feb 28 15:23:20 KinkPad-K450 multipassd[8597]: Unable to determine subnet for the mpqemubr0 subnet
Feb 28 15:23:21 KinkPad-K450 dnsmasq[8703]: started, version 2.75 cachesize 150
Feb 28 15:23:21 KinkPad-K450 dnsmasq[8703]: compile time options: IPv6 GNU-getopt DBus i18n IDN DHCP DHCPv6 no-Lua TFTP conntrack ipset
Feb 28 15:23:21 KinkPad-K450 dnsmasq-dhcp[8703]: DHCP, IP range 10.197.20.2 -- 10.197.20.254, lease time infinite
Feb 28 15:23:21 KinkPad-K450 dnsmasq-dhcp[8703]: DHCP, sockets bound exclusively to interface mpqemubr0
Feb 28 15:23:21 KinkPad-K450 dnsmasq[8703]: reading /etc/resolv.conf
Feb 28 15:23:21 KinkPad-K450 dnsmasq[8703]: using nameserver 127.0.0.53#53
Feb 28 15:23:21 KinkPad-K450 dnsmasq[8703]: read /etc/hosts - 7 addresses
Feb 28 15:23:21 KinkPad-K450 dnsmasq[8703]: cannot read /var/snap/multipass/common/data/multipassd/network/dnsmasq.hosts: No such file 
Feb 28 15:23:21 KinkPad-K450 multipassd[8597]: "No such interface 'org.freedesktop.DBus.Properties' on object at path /org/freedesktop/
Feb 28 15:23:21 KinkPad-K450 multipassd[8597]: Host cloud-images.ubuntu.com not found
Feb 28 15:23:21 KinkPad-K450 systemd[1]: snap.multipass.multipassd.service: Main process exited, code=exited, status=1/FAILURE
Feb 28 15:23:21 KinkPad-K450 systemd[1]: snap.multipass.multipassd.service: Failed with result 'exit-code'.
Feb 28 15:23:21 KinkPad-K450 systemd[1]: snap.multipass.multipassd.service: Service hold-off time over, scheduling restart.
Feb 28 15:23:21 KinkPad-K450 systemd[1]: snap.multipass.multipassd.service: Scheduled restart job, restart counter is at 4.
Feb 28 15:23:21 KinkPad-K450 systemd[1]: Stopped Service for snap application multipass.multipassd.
Feb 28 15:23:21 KinkPad-K450 systemd[1]: Started Service for snap application multipass.multipassd.
Feb 28 15:23:21 KinkPad-K450 multipassd[8758]: Unable to determine subnet for the mpqemubr0 subnet
Feb 28 15:23:21 KinkPad-K450 dnsmasq[8856]: started, version 2.75 cachesize 150
Feb 28 15:23:21 KinkPad-K450 dnsmasq[8856]: compile time options: IPv6 GNU-getopt DBus i18n IDN DHCP DHCPv6 no-Lua TFTP conntrack ipset
Feb 28 15:23:21 KinkPad-K450 dnsmasq-dhcp[8856]: DHCP, IP range 10.197.20.2 -- 10.197.20.254, lease time infinite
Feb 28 15:23:21 KinkPad-K450 dnsmasq-dhcp[8856]: DHCP, sockets bound exclusively to interface mpqemubr0
Feb 28 15:23:21 KinkPad-K450 dnsmasq[8856]: reading /etc/resolv.conf
Feb 28 15:23:21 KinkPad-K450 dnsmasq[8856]: using nameserver 127.0.0.53#53
Feb 28 15:23:21 KinkPad-K450 dnsmasq[8856]: read /etc/hosts - 7 addresses
Feb 28 15:23:21 KinkPad-K450 dnsmasq[8856]: cannot read /var/snap/multipass/common/data/multipassd/network/dnsmasq.hosts: No such file 
Feb 28 15:23:21 KinkPad-K450 multipassd[8758]: "No such interface 'org.freedesktop.DBus.Properties' on object at path /org/freedesktop/
Feb 28 15:23:21 KinkPad-K450 multipassd[8758]: Host cloud-images.ubuntu.com not found
Feb 28 15:23:21 KinkPad-K450 systemd[1]: snap.multipass.multipassd.service: Main process exited, code=exited, status=1/FAILURE
Feb 28 15:23:21 KinkPad-K450 systemd[1]: snap.multipass.multipassd.service: Failed with result 'exit-code'.
Feb 28 15:23:22 KinkPad-K450 systemd[1]: snap.multipass.multipassd.service: Service hold-off time over, scheduling restart.
Feb 28 15:23:22 KinkPad-K450 systemd[1]: snap.multipass.multipassd.service: Scheduled restart job, restart counter is at 5.
Feb 28 15:23:22 KinkPad-K450 systemd[1]: Stopped Service for snap application multipass.multipassd.
Feb 28 15:23:22 KinkPad-K450 systemd[1]: Started Service for snap application multipass.multipassd.
Feb 28 15:23:22 KinkPad-K450 multipassd[8918]: Unable to determine subnet for the mpqemubr0 subnet
Feb 28 15:23:22 KinkPad-K450 dnsmasq[9019]: started, version 2.75 cachesize 150
Feb 28 15:23:22 KinkPad-K450 dnsmasq[9019]: compile time options: IPv6 GNU-getopt DBus i18n IDN DHCP DHCPv6 no-Lua TFTP conntrack ipset
Feb 28 15:23:22 KinkPad-K450 dnsmasq-dhcp[9019]: DHCP, IP range 10.197.20.2 -- 10.197.20.254, lease time infinite
Feb 28 15:23:22 KinkPad-K450 dnsmasq-dhcp[9019]: DHCP, sockets bound exclusively to interface mpqemubr0
Feb 28 15:23:22 KinkPad-K450 dnsmasq[9019]: reading /etc/resolv.conf
Feb 28 15:23:22 KinkPad-K450 dnsmasq[9019]: using nameserver 127.0.0.53#53
Feb 28 15:23:22 KinkPad-K450 dnsmasq[9019]: read /etc/hosts - 7 addresses
Feb 28 15:23:22 KinkPad-K450 dnsmasq[9019]: cannot read /var/snap/multipass/common/data/multipassd/network/dnsmasq.hosts: No such file 
Feb 28 15:23:22 KinkPad-K450 multipassd[8918]: "No such interface 'org.freedesktop.DBus.Properties' on object at path /org/freedesktop/
Feb 28 15:23:22 KinkPad-K450 multipassd[8918]: Host cloud-images.ubuntu.com not found
Feb 28 15:23:22 KinkPad-K450 systemd[1]: snap.multipass.multipassd.service: Main process exited, code=exited, status=1/FAILURE
Feb 28 15:23:22 KinkPad-K450 systemd[1]: snap.multipass.multipassd.service: Failed with result 'exit-code'.
Feb 28 15:23:22 KinkPad-K450 systemd[1]: snap.multipass.multipassd.service: Service hold-off time over, scheduling restart.
Feb 28 15:23:22 KinkPad-K450 systemd[1]: snap.multipass.multipassd.service: Scheduled restart job, restart counter is at 6.
Feb 28 15:23:22 KinkPad-K450 systemd[1]: Stopped Service for snap application multipass.multipassd.
Feb 28 15:23:22 KinkPad-K450 systemd[1]: snap.multipass.multipassd.service: Start request repeated too quickly.
Feb 28 15:23:22 KinkPad-K450 systemd[1]: snap.multipass.multipassd.service: Failed with result 'exit-code'.
Feb 28 15:23:22 KinkPad-K450 systemd[1]: Failed to start Service for snap application multipass.multipassd.

```

**Comment by Saviq** [comment]
This should be fixed on edge, #611 made our networking much more resilient. The `edge` channel has that fix already, we'll be releasing a new version with it soon.

**Comment by Saviq** [comment]
Duplicate of #555
